pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -337,3 +375,52 @@ func gitRefInfo(repo git.Repository, dir, ref string) (buildapi.GitRefInfo, erro
 	info.Message = lines[4]
 	return info, nil
 }
+
+// WaitForBuildComplete waits for a build identified by the name to complete
+func WaitForBuildComplete(c osclient.BuildInterface, name string) error {
+	isOK := func(b *buildapi.Build) bool {
+		return b.Status.Phase == buildapi.BuildPhaseComplete
+	}
+	isFailed := func(b *buildapi.Build) bool {
```

Comment:
@kargakis also the server function is meant to give you logs when the build is running, complete or failed whereas the function I have cares just about Complete/Failed/Cancelled... so I guess this is a different function.... But I agree that we should have something generic we can re-use.... WaitForBuild(BuildInterface, okCondition, failedCondition).... I think we already have this in extended tests, need to move that to Origin.